prime-router/src/main/kotlin/history/azure/SubmissionsFacade.kt

```diff
@@ -116,30 +122,56 @@ class SubmissionsFacade(
      * @return Report details
      */
     fun findDetailedSubmissionHistory(
+        txn: DataAccessTransaction,
+        id: UUID?,
         action: Action,
     ): DetailedSubmissionHistory? {
-        // This assumes that ReportFileFunction.authSingleBlocks has already run, and has checked that the
-        // sendingOrg is good.  If that assumption is incorrect, die here.
-        assert(action.sendingOrg != null && action.actionName == TaskAction.receive)
-        val submission = dbSubmissionAccess.fetchAction(
-            action.actionId,
-            action.sendingOrg,
-            DetailedSubmissionHistory::class.java
-        )
-        submission?.actionsPerformed?.add(action.actionName)
-
-        // Submissions with a report ID (means had no errors) can have a lineage
-        submission?.reportId?.let {
-            val relatedSubmissions = dbSubmissionAccess.fetchRelatedActions(
-                UUID.fromString(it),
+        if (id == null) {
```

Comment:
this is a nit but we have a lot of "id" object floating around. It might be worthwhile to specify exactly which "id" we're dealing with